src/config.rs

```diff
@@ -508,6 +611,32 @@ impl TiKvConfig {
                 Path::new(&self.storage.data_dir).join("backup").display()
             );
         }
+
+        let kv_db_path = try!(config::canonicalize_sub_path(&self.storage.data_dir, "db"));
+        if kv_db_path == self.storage.raft_db_path {
```

Comment:
What you check here is the string, but they are path actually. For example, `"data" != "data/../data"`, but they can point to the same directory.